# Evaluating if method 1 (novelty in recombination) from Veugelers paper is feasible

Novelty in recombination: Determine if a pairwise combination of USPTO classes of a patent has occured previously, or is it the first (and thus novel) occurance

In [1]:
import pandas as pd

In [2]:
#take in data
master = pd.read_csv('USDesign.csv')
master.head()

patent_number    app_date    class mainclass
0       D257924  1980-01-14  D06/573       D06
1       D258382  1980-01-17  D23/214       D23
2       D258383  1980-01-17  D23/214       D23
3       D258571  1980-04-03  D09/560       D09
4       D258650  1980-06-09  D12/410       D12

Based off full classification

In [4]:
num_pats = master.patent_number.nunique()

In [5]:
num_pats

415134

In [6]:
num_pats_multiclass = master[master.duplicated(subset=['patent_number'], keep=False)].patent_number.nunique()

In [7]:
num_pats_multiclass

113248

In [8]:
((num_pats - num_pats_multiclass)/num_pats) * 100

72.72013373994903

~~There are 342,800 data points representing 312,050 unique patent numbers. 27,448 of these patents have more than one class associated with them at the group level, resulting in 90% of the patent population only having one class. Based on these metrics, it is **not** appropriate to use novelty in recombination as a measure.~~

~~**Important** The statement above was determined during the intial data query, which was only requesting the group level classification.~~

~~There are 358,321 data data points representing 254,113 unique patent numbers. More work will have to be done to understand why there is a discrepancy between the first and second totals of unique patent numbers. 72,758 Of these patents have classes associated with them, resulting in 71.36% of the patents only havine one class. Based on these metrics, it is not appropriate to use novelty in recombination as a measure.~~

The lastest pull represents 415,134 unique patent numbers. Of these about 72% of the patents only have one class associated with them. Based off this, novelty in recombination does not seem like the best method